torch/_inductor/utils.py

```diff
@@ -977,7 +977,15 @@ def use_triton_template(layout, *, enable_int32=False):
     )
 
 
-def use_cutlass_template(layout):
+def use_cutlass_template(layout, m, n, k):
+    gemm_size = -1
+    try:
+        # this fails if it's dynamic
+        gemm_size = int(m * n * k)
+    except Exception:
+        pass
```

Comment:
Rather than erroring you can use `V.graph.sizevars.size_hint`.

Reply:
Changed that here. I think I should also use it in the PR above it in this stack ( about tolerating dynamic shapes ) to determine GEMM dimensions.